### PR TITLE
Add the timezone information.

### PR DESCRIPTION
**Pull request by Shylock-Hg**
Follow https://github.com/vesoft-inc/nebula-graph/pull/348

### CODE REVIEW

File: src/common/time/TimezoneInfo.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    MUST_USE_RESULT Status load() {
+        const auto &it = tzdb.find(name_);
+        if (it == tzdb.end()) {
+            return Status::Error("Not supported timezone `%s'.", name_.c_str());
+        }
+#if DCHECK_IS_ON()
+        valid_ = true;
+#endif
+        return Status::OK();
```
Comment (laura-ding) [inline src/common/time/TimezoneInfo.h:28]: need to init `name_` and `info_`.

and add tests.

File: src/common/time/Timezone [hunk 2]
```diff
@@ -0,0 +1,529 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+// Generated by boost::date_time timezone resources and https://github.com/Shylock-Hg/tzDBTransformer
```
Comment (laura-ding) [inline src/common/time/Timezone:7]: why not use the boost directly

Comment (Shylock-Hg) [reply src/common/time/Timezone:7]: Good catch. I don't know we have boost in 3rd-party.

File: src/common/time/TimezoneInfo.h [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    MUST_USE_RESULT Status load() {
+        zoneInfo_ = tzdb.time_zone_from_region(name_);
+        if (zoneInfo_ == nullptr) {
+            return Status::Error("Not supported timezone `%s'.", name_.c_str());
+        }
+        return Status::OK();
+    }
+
+    // offset in seconds
+    uint32_t utcOffset() const {
```
Comment (laura-ding) [inline src/common/time/TimezoneInfo.h:40]: unsigned?

File: src/common/time/TimezoneInfo.h [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    static MUST_USE_RESULT Status init() {
+        try {
+            tzdb.load_from_file(kTzDbFile);
+        } catch (const std::exception &e) {
+            return Status::Error("%s", e.what());
+        }
+        return Status::OK();
+    }
+
+    MUST_USE_RESULT Status load_from_db(const std::string &region) {
```
Comment (laura-ding) [inline src/common/time/TimezoneInfo.h:31]: function name format

File: src/common/time/TimezoneInfo.h [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        zoneInfo_ = tzdb.time_zone_from_region(region);
+        if (zoneInfo_ == nullptr) {
+            return Status::Error("Not supported timezone `%s'.", name_.c_str());
+        }
+        return Status::OK();
+    }
+
+    MUST_USE_RESULT Status parse_posix_timezone(const std::string &posixTimezone) {
```
Comment (laura-ding) [inline src/common/time/TimezoneInfo.h:39]: ditto

File: src/common/time/TimezoneInfo.h (hunk 3 above)
Comment (Shylock-Hg) [reply src/common/time/TimezoneInfo.h]: Had fixed.

File: src/common/time/TimezoneInfo.h (hunk 4 above)
Comment (Shylock-Hg) [reply src/common/time/TimezoneInfo.h]: Fixed.

File: src/common/time/TimezoneInfo.h (hunk 5 above)
Comment (Shylock-Hg) [reply src/common/time/TimezoneInfo.h]: Fixed.

File: src/common/time/TimezoneInfo.h [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    // offset in seconds
+    int32_t utcOffset() const {
+        return DCHECK_NOTNULL(zoneInfo_)->base_utc_offset().total_seconds();
+    }
+
+private:
+    static constexpr char kTzDbFile[] = "share/resources/date_time_zonespec.csv";
+    static ::boost::local_time::tz_database tzdb;
+
+    ::boost::shared_ptr<::boost::date_time::time_zone_base<::boost::posix_time::ptime, char>>
```
Comment (yixinglu) [inline src/common/time/TimezoneInfo.h:63]: prefer `std::shared_ptr`

File: src/common/time/TimezoneInfo.h [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        }
+        return Status::OK();
+    }
+
+    std::string stdZoneName() const {
+        return DCHECK_NOTNULL(zoneInfo_)->std_zone_name();
+    }
+
+    // offset in seconds
+    int32_t utcOffset() const {
```
Comment (yixinglu) [inline src/common/time/TimezoneInfo.h:55]: utcOffsetInSecs

File: src/common/time/TimezoneInfo.h [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        return DCHECK_NOTNULL(zoneInfo_)->base_utc_offset().total_seconds();
+    }
+
+private:
+    static constexpr char kTzDbFile[] = "share/resources/date_time_zonespec.csv";
```
Comment (yixinglu) [inline src/common/time/TimezoneInfo.h:60]: It's better to let user pass the path in when construct or init Timezone instance. If they don't install the resource file and use the Timezone, they will encounter some errors. In addition, users couldn't use their own timezone resource files.

File: src/common/time/TimezoneInfo.h (hunk 6 above)
Comment (Shylock-Hg) [reply src/common/time/TimezoneInfo.h:63]: The line 33 return ::boost::shared_ptr.

Comment (Shylock-Hg) [reply src/common/time/TimezoneInfo.h:63]: And it share the ownership with the tzdb.

File: src/common/time/TimezoneInfo.h (hunk 7 above)
Comment (Shylock-Hg) [reply src/common/time/TimezoneInfo.h]: Fixed.

File: src/common/time/TimezoneInfo.h (hunk 8 above)
Comment (Shylock-Hg) [reply src/common/time/TimezoneInfo.h:60]: Could make it gflags.

**Review by CPWstatic — APPROVED** [review]
LGTM.

**Review by dutor — APPROVED** [review]
Well done.

**Pull request merged**